cmd/minishift/cmd/root.go

```diff
@@ -213,14 +222,37 @@ func initializeProfile() string {
 		activeProfile string
 	)
 
+	// If `minishift profile set` or `minishift start --profile` command is used.
 	for i, arg := range os.Args {
+		// This will match if `--profile` flag is used
 		if arg == "--"+profileFlag {
+			// This make sure if user specify `--profile` flag without any name
+			// it should not panic with out of index error
+			if len(os.Args) <= i+1 {
+				break
+			}
 			profileName = os.Args[i+1]
 			if !cmdUtil.IsValidProfileName(profileName) {
 				atexit.ExitWithMessage(1, "Profile names must consist of alphanumeric characters only.")
 			}
 			break
 		}
+		// This will match if we used profile or it's alias commands
+		if arg == profileFlag || arg == profileAlias[0] || arg == profileAlias[1] {
+			// This make sure if user specify profile command without any subcommand then
+			// it should not panic with out of index error
+			if len(os.Args) <= i+2 {
+				break
+			}
+			// For use cases when minishift profile set PROFILE_NAME is used
+			if os.Args[i+1] == profileSetFlag {
+				profileName = os.Args[i+2]
+				if !cmdUtil.IsValidProfileName(profileName) {
+					atexit.ExitWithMessage(1, "Profile names must consist of alphanumeric characters only.")
+				}
+			}
+			break
```

Comment:
These code change is completely changing the behavior of initializeProfile which might not be intended.  Just take example of below commands
* minishift profile list --profile xyz
* minishift --profile xyz profile list
* minishift profile delete xyz 
InitilizeProfile behaves differently around these commands which is not expected IMO. 

Also it fails for `$ minishift --profile xyz profile list` i.e. it creates the `xyz` profile.

Reply:
@LalatenduMohanty Good catch 👍 , I need to implement another flag to make sure if `profile` and `--profile` args are there then `profile` always take preference.